### ISSUE TITLE
`no-unnecessary-polyfills`: False negatives for `Promise` imports with core-js-compat 3.50.0

### ISSUE THREAD

**Issue by snitin315** [sindresorhus/eslint-plugin-unicorn#3606]
`no-unnecessary-polyfills` no longer reports `core-js/stable/promise` (or any `Promise` polyfill import) for **any** target, so its own test suite fails on a fresh install:

```
no-unnecessary-polyfills › invalid(6): require("core-js/stable/promise")

  Should have 1 error but had 0: []
```

**Rule:** `no-unnecessary-polyfills`

**Cause:** `core-js-compat@3.50.0` (published 2026-08-05) emptied the support data for `es.promise.try` and `esnext.promise.try` — in 3.49.0 both listed `{"chrome": "128", "node": "23.0", "safari": "18.2", "firefox": "134", ...}`, in 3.50.0 both are `{}`. They are the only two keys in `data.json` that went from populated to empty between the two releases.

Since `es.promise.try` is now considered unavailable everywhere, the rule always sees an unavailable feature in the import and stays silent. The dependency range is `"core-js-compat": "^3.49.0"`, so new installs pick 3.50.0 up.

**Examples where the issue appears:**

```js
// Expected: reported (all features available in Node > 24). Actual: not reported.
require('core-js/stable/promise');
// options: [{targets: 'node >24'}]

// Same for any Promise entry point, e.g.
import 'core-js/features/promise';
```

Reproducible without ESLint:

```js
const compat = require('core-js-compat');
compat({targets: 'node >24', modules: ['core-js/stable/promise']}).list;
// 3.49.0 -> []
// 3.50.0 -> ['es.promise.try']
```

Pinning `core-js-compat` to 3.49.0 makes `test/no-unnecessary-polyfills.js` pass 73/73.

This looks like a core-js data regression (`Promise.try` has shipped in Chrome 128+, Safari 18.2+, Firefox 134+, Node 23+), so it likely needs fixing upstream filing here since it currently breaks the plugin's own tests. Spotted via [ESLint's ecosystem test suite](https://github.com/eslint/eslint/actions/runs/31037017789/job/92411529164?pr=20503#step:7:24); no ESLint change is involved.

**Comment by zloirock** [comment]
> This looks like a core-js data regression

Nope. [Just `Promise.try` changed its behaviour.](https://github.com/tc39/ecma262/pull/3883) Please at least read the changelog before writing something like this.
